controllers/vrg_volsync.go

```diff
@@ -92,61 +92,9 @@ func (v *VRGInstance) reconcileVolSyncAsPrimary() (requeue bool) {
 		return
 	}
 
-	finalSyncPrepareCount := 0
-
-	// First time: Add all VolSync PVCs to the protected PVC list and set their ready condition to initializing
 	for _, pvc := range v.volSyncPVCs {
-		newProtectedPVC := &ramendrv1alpha1.ProtectedPVC{
-			Name:               pvc.Name,
-			ProtectedByVolSync: true,
-			StorageClassName:   pvc.Spec.StorageClassName,
-			Labels:             pvc.Labels,
-			AccessModes:        pvc.Spec.AccessModes,
-			Resources:          pvc.Spec.Resources,
-		}
-
-		protectedPVC := v.findProtectedPVC(pvc.Name)
-		if protectedPVC == nil {
-			protectedPVC = newProtectedPVC
-			v.instance.Status.ProtectedPVCs = append(v.instance.Status.ProtectedPVCs, *protectedPVC)
-		} else if !reflect.DeepEqual(protectedPVC, newProtectedPVC) {
-			newProtectedPVC.DeepCopyInto(protectedPVC)
-		}
-
-		// Not much need for VolSyncReplicationSourceSpec anymore - but keeping it around in case we want
-		// to add anything to it later to control anything in the ReplicationSource
-		rsSpec := ramendrv1alpha1.VolSyncReplicationSourceSpec{
-			ProtectedPVC: *protectedPVC,
-		}
-
-		if v.instance.Spec.PrepareForFinalSync {
-			prepared, err := v.volSyncHandler.PreparePVCForFinalSync(pvc.Name)
-			if err != nil || !prepared {
-				requeue = true
-
-				continue
-			}
-
-			finalSyncPrepareCount++
-		}
-
-		// reconcile RS and if runFinalSync is true, then one final sync will be run
-		finalSyncComplete, rs, err := v.volSyncHandler.ReconcileRS(rsSpec, v.instance.Spec.RunFinalSync)
-		if err != nil { //nolint:gocritic
-			v.log.Info(fmt.Sprintf("Failed to reconcile VolSync Replication Source for rsSpec %v. Error %v",
-				rsSpec, err))
-
-			setVRGConditionTypeVolSyncRepSourceSetupError(&protectedPVC.Conditions, v.instance.Generation,
-				"VolSync setup failed")
-
-			requeue = true
-		} else if rs == nil {
-			requeue = true
-		} else {
-			setVRGConditionTypeVolSyncRepSourceSetupComplete(&protectedPVC.Conditions, v.instance.Generation, "Ready")
-		}
-
-		if v.instance.Spec.RunFinalSync && !finalSyncComplete {
+		requeuePVC := v.reconcilePVCAsVolSyncPrimary(pvc)
```

Comment:
(note) minor refactor to address linter issues due to new code (cyclomatic complexity)